File: src/WorkBC.Web/ClientApp/projects/jb-account/src/app/models/register.model.ts
import { JobSeekerFlags } from './job-seeker-flags.model';
import { UserBase } from '../../../../jb-lib/src/public-api';

export interface Register extends UserBase {
    countryId: number;
    provinceId: number;
    city: string;
    locationId: number;
    securityQuestionId: number;
    securityAnswer: string;

    jobSeekerFlags: JobSeekerFlags;
}

export class RegisterModel implements Register {
    username: string;

    constructor(
        public email: string = '',
        public password: string = '',
        public firstName: string = '',
        public lastName: string = '',
        public countryId: number = 0,
        public provinceId: number = 0,
        public city: string = '',
        public locationId: number = 0,
        public securityQuestionId: number = 0,
        public securityAnswer: string = '',
        public jobSeekerFlags: JobSeekerFlags = null
    ) {
        if (email) {
            this.username = email;
        }
    }
}

export interface ForgotPasswordModel {
  email: string;
  token: string;
  newPassword?: string;
}

File: src/WorkBC.Web/Models/RegisterModel.cs
using WorkBC.Data.Model.JobBoard;

namespace WorkBC.Web.Models
{
    public interface IRegisterModel: ILoginModel, IUserBase
    {
        int CountryId { get; set; }
        int ProvinceId { get; set; }
        string City { get; set; }
        int LocationId { get; set; }
        int SecurityQuestionId { get; set; }
        string SecurityAnswer { get; set; }
        JobSeekerFlags JobSeekerFlags { get; set; }
    }

    public class RegisterModel : IRegisterModel
    {
        public string Email { get; set; }
        public string Username { get; set; }        
        public string Password { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }        
        public int CountryId { get; set; }
        public int ProvinceId { get; set; }
        public string City { get; set; }
        public int LocationId { get; set; }
        public int SecurityQuestionId { get; set; }
        public string SecurityAnswer { get; set; }
        public JobSeekerFlags JobSeekerFlags { get; set; }
    }
}